src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java

```diff
@@ -5,7 +5,7 @@
 import javafx.beans.property.ObjectProperty;
 import javafx.beans.property.SimpleObjectProperty;
 
-public class DeadlineTask extends Task implements Comparable<DeadlineTask> {
+public class DeadlineTask extends Task implements Comparable<Task> {
 
```

Comment:
OKay. I see where the problem is. I think the best solution is to remove Comparable from Task and move it to Deadline and FloatingTask.